staging/src/k8s.io/kubectl/pkg/cmd/get/get.go

```diff
@@ -813,7 +813,6 @@ func (o *GetOptions) printGeneric(r *resource.Result) error {
 		}
 		if listMeta, err := meta.ListAccessor(obj); err == nil {
 			list.Object["metadata"] = map[string]interface{}{
-				"selfLink":        listMeta.GetSelfLink(),
```

Comment:
this is client-side, so it should faithfully echo what the server returns. make this conditional:

```go
if selfLink := listMeta.GetSelfLink(); len(selfLink) > 0 {
  list.Object["metadata"]["selfLink"] = selfLink
}
```

Reply:
I think I don't agree with this case. As long as for pager it makes sense to me, here it's purely about printing some metadata (selflink in particular). Printing that from kubectl doesn't make sense to me, given this feature is no longer supporting (and if you have very old server that is still setting selflinks, you can always get it via -o json or -o yaml or sth).

Reply:
But we're not setting all fields anyway, right?

Reply:
OK - I reverted this commit.

Though I'm planning to open a separate PR for that once this one merges - I don't think this brings any value...

Reply:
sounds good. I'd also be ok with making `selfLink` here optional based on whether the server returned a non-empty value